### ISSUE TITLE
Add detailed diffs of changes made to secrets

### ISSUE THREAD

**Issue by rohan-chaturvedi** [phasehq/console#274]
## Is your feature request related to a problem?

Making changes to secrets currently shows a "You have undeployed changes to this environment." warning banner and an amber highlight to affected secrets, but it can be difficult to determine what changes were made other than which secrets are affected.

[2024-06-16 20-44-40.webm](https://github.com/phasehq/console/assets/6710327/0c8aa0e9-8848-4130-a857-92ad9afa1a2c)

## Describe the solution you'd like
Clicking the warning banner should show a modal with details of secrets affected, with diffs of specific properties changed, such as tags, comments, etc. For example:
* Modified `GOOGLE_CLIENT_ID` 
  * changed key
  * added tag
* Added`TEST`
* Modified `SESSION_COOKIE_DOMAIN`
  * changed comment

Ideally, something inspired by VSCode's diffs would be great:

![image](https://github.com/phasehq/console/assets/6710327/400f5aaf-73f5-4581-8da4-5d6f388f080f)

**Comment by harshith-1008** [comment]
hey, I would like to work on this issue. @rohan-chaturvedi

**Comment by rohan-chaturvedi** [comment]
> hey, I would like to work on this issue. @rohan-chaturvedi

Great, its all yours!

**Comment by harshith-1008** [comment]
<img width="1470" alt="Screenshot 2024-07-11 at 12 09 50 PM" src="https://github.com/phasehq/console/assets/123820448/8dd7326d-cce2-48cb-a30c-36515237b03c">
<img width="1470" alt="Screenshot 2024-07-11 at 12 14 03 PM" src="https://github.com/phasehq/console/assets/123820448/1bd56ec9-ca02-4f14-b567-04308810af25">



Hey @rohan-chaturvedi just wanted a little clarification about the UI, for testing I have created a dropdown ui, should I make a separate pop up screen/page for the changes or should I just try to make the current dropdown look cleaner?

**Comment by rohan-chaturvedi** [comment]
Hey @harshith-1008 , this is a good start! However I think we should avoid this "accordion" style behavior, since it shifts the entire rest of the UI down. Instead, how about trying a modal dialog? We have a `GenericDialog` component you could use: https://github.com/phasehq/console/blob/main/frontend/components/common/GenericDialog.tsx

**Comment by harshith-1008** [comment]
thanks for the feedback, ill try to implement it.

**Comment by rohan-chaturvedi** [comment]
> @rohan-chaturvedi I tried using genericDialogue component, but it expects button variant props, if I pass button variant then the alert changes to green box around the text. I have changed the component such that it accepts an empty string as props for genericDialogue, is it fine? <img alt="Screenshot 2024-07-25 at 1 10 17 PM" width="1470" src="https://private-user-images.githubusercontent.com/123820448/352009570-7fd30231-1676-4e7f-9e54-948e2fec1dbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4OTQ4ODYsIm5iZiI6MTcyMTg5NDU4NiwicGF0aCI6Ii8xMjM4MjA0NDgvMzUyMDA5NTcwLTdmZDMwMjMxLTE2NzYtNGU3Zi05ZTU0LTk0OGUyZmVjMWRiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNVQwODAzMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NzExNWQ2N2Y2ZjYzMGI5NWRiMGY0ZWI0NzNkODU3YjI5ODU2NDdhMjMxOGJmODEwMDFiMzkzZGIxYmFhYWVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GdY9gYCg4WCveRnx28gAYsGFtqmH4SxfBtcAXhKADFw">

@harshith-1008 that should be fine, just make sure that its not a breaking change for other components that use it. I'll take a look in detail once you have a PR. Once thing you could try is adding an `i` icon in an `outline` or `secondary` button as the modal trigger rather than the entire alert text :thinking:

**Comment by harshith-1008** [comment]
<img width="1202" alt="Screenshot 2024-07-27 at 10 47 12 PM" src="https://github.com/user-attachments/assets/c3372e5b-62e6-4939-975d-61047c66c5c4">
<img width="1202" alt="Screenshot 2024-07-27 at 10 47 23 PM" src="https://github.com/user-attachments/assets/6285155b-576f-4956-bfea-e48ffa51fb7d">


@rohan-chaturvedi I hope this UI is fine.

**Comment by rohan-chaturvedi** [comment]
This is a great start! A few suggestions:
* Add a bit more vertical spacing between items
* Newly added secrets should show as "Added `CHECKING`" rather than "Modified **_'Added CHECKING'_**"
* Use `MONOSPACE` fonts for key names
* Highlight the specific field that was changed, perhaps by using a higher font weight like `font-medium` or a different font color, (maybe `font-mono` as well?)
* Explore the [FontAwesome Icons](https://react-icons.github.io/react-icons/icons/fa/) and try and use some icons to denote modifications vs additions

**Comment by harshith-1008** [comment]
<img width="1202" alt="Screenshot 2024-07-28 at 12 34 14 AM" src="https://github.com/user-attachments/assets/263fdfac-8a02-4179-b8c5-849d1d8dbecd">

@rohan-chaturvedi  is this ui okay?

**Comment by rohan-chaturvedi** [comment]
That looks awesome! A few notes:
* Only use `font-mono` for key names or field names. For example: "Modified `AWS_SECRET_ACCESS_KEY`"
* Make sure the color-coding logic is consistent. Green makes sense for a new secret, but for modified secrets there's both blue and yellow. I'd suggest just picking one (yellow makes more sense IMO)
* Could we show details of the changes here? For example:

* Modified `AWS_SECRET_ACCESS_KEY`
  * Changed `value` 
    * `aCRAMarEbFC3Q5c24pi7AVMIt6TaCfHeFZ4KCf/a` -> `JBgoiFGNdbutADANDPmjp[OMKOJHduafdKAD`
    
 You could use the secret history UI as a baseline for this as well:
 
 
![image](https://github.com/user-attachments/assets/597641e8-5310-44b1-88eb-7582f1694a18)

**Comment by rohan-chaturvedi** [comment]
@harshith-1008 The [SecretPropertyDiffs](https://github.com/phasehq/console/blob/main/frontend/components/environments/secrets/SecretPropertyDiffs.tsx) component could be useful for rendering details of changes

**Comment by harshith-1008** [comment]
@rohan-chaturvedi  just one small clarification, the font-mono should be just on keys name right? not the properties name right?

**Comment by rohan-chaturvedi** [comment]
Yep, I think that will work well

**Comment by harshith-1008** [comment]
<img width="1026" alt="Screenshot 2024-07-31 at 11 00 50 AM" src="https://github.com/user-attachments/assets/8666f00a-0199-4c78-8dd4-65f6cdf0366e">

@rohan-chaturvedi Is this UI fine?

**Comment by rohan-chaturvedi** [comment]
@harshith-1008 This is good progress! The UI still needs a bit of work though. Could you tag me on the PR when its ready? It will be easier to suggest changes to the styles.

**Comment by iamppborah** [comment]
Hi @rohan-chaturvedi, this is good to be closed ?

**Comment by rohan-chaturvedi** [comment]
> Hi [@rohan-chaturvedi](https://github.com/rohan-chaturvedi), this is good to be closed ?

Yep. I do want to improve the diff view and UI in general, but we can create a new issue with those specific requirements. Closing this.
